### DISCUSSION TITLE
How to make search more intuitive and efficient

### DISCUSSION THREAD

**Discussion by ibrahimrahman** [elementary/files#1920] [Ideas]
## What Happened

Searching for the a file/folder is incredibly inefficient as it is located at the very bottom of the list, which is filled with irrelevant entries. See screenshot:

![Screenshot-search](https://user-images.githubusercontent.com/5392273/130155419-fbfb9e2a-6bef-4ac1-b021-d18b7768fe88.png)

I just wanted to quickly navigate to the Github folder in my home directory.

## Expected Behavior

The search results from the current directory should be given the highest priority. 

What would really improve search is if the entries in the files window itself could be filtered. This would help with easier comparisons of/operations on the entries, especially when looking for multiple items with a similar name. As of now, the search funtion in Files is incredibly basic and leaves a lot to be desired, especially in comparison to Nautilus and other file managers

## Steps to Reproduce

1. Just search for something

## Platform Information

elementary OS 6 Odin

**Comment by jeremypw** [comment]
I think the "In this folder" category used to be at the top so this looks like a regression.  I agree that this category should be at the top.  Filtering the view is on the wishlist but is not trivial to implement.

**Comment by abodedis** [comment]
I haven't used search since the beginning as I was well aware of its limitations.  Thanks for reminding me that elementary OS has a file search :)

In the future, it could be useful to bundle elementary with a default File Search App.
Several nice ones exist that are capable of instantaneous performance (not including indexing time), like [FSearch](http://cboxdoerfer.github.io/fsearch/), among many others.

**Comment by jeremypw** [comment]
Any app that pre-indexes the files is going to easily outperform Files, which has to recursively search the current folder every time.  I would like some way of saving the current search results perhaps in a view, at least temporarily since at the moment the search results are lost as soon as you click on one of them.

**Comment by lazaroofarrill** [comment]
I'd love to at least have the option of disabling recursive search. Search is even worst than in Hera. Looking for files recursively is way less common than looking for the files in your current folder. Navigation that used to take no more than 4 seconds without even looking at my screen, now requires all my attention and to manually scroll through all my files, since waiting for a search result is not even an option.

**Reply by idovecer** [reply]
Yes, me too. Most of the time I just quickly need to get positioned on some folder or file, don't use very often nor need recursive search most of the time.

For example I have folder with 200-300 subfolders, don't wanna scroll nor use mouse, just like to type 2-3 letters to get fast focus for typed/searched folder and click enter on keyboard to open this folder.
Or type 2-3 letters to get focus on some file and then click enter to open this file. 

I think that **recursive search and finding something fast inside** the current folder must be **two divided functions**.

**Reply by jeremypw** [reply]
> Looking for files recursively is way less common than looking for the files in your current folder.

Not for me it isn't.  Generally speaking,. I can find something in the current folder quicker by scanning it by eye (with a suitable sort order).

> I think that recursive search and finding something fast inside the current folder must be two divided functions.
Not sure what you mean by this.  If you mean there should be separate UX for these operations then that makes the interface more complicated.

I do agree however that getting results for the current folder needs to be much faster.

**Reply by idovecer** [reply]
Yes, I agree with you that you can find the folder faster by scanning it by eye if you have 5-10 subfolders/files inside of the folder. But that's not the case when you have 50+ subfolders.
I tried to compare again PCManFM, Thunar and Files, I have folder with 264 subfolders. And every folder name is name of my client, I just need to type first 2-3 letters of clients name and press enter and I'm inside of this subfolder in less than a second by using just 3-4 keys.

Using elementary Files it isn't possible to be that fast. You need to use combination of keyboard + mouse (that's slow), or you can do it by just using keyboard and then you need to press 10-20 keys (down key) for the same task.

Search and to get position of folder/file should be in my opinion two different functions or somehow implement fast positioning on wanted subfolder/file.

**Comment by jeremypw** [comment]
I agree.  I have been intending to change things so that search results in "This Folder" are immediately available whether or not the recursive search is finished.  Just a question of finding the time.

**Comment by jeremypw** [comment]
I think this should be converted to a discussion and then issues raised about more focussed issues/solutions.
